crates/router/src/connector/braintree/transformers.rs

```diff
@@ -255,234 +544,1268 @@ impl From<BraintreePaymentStatus> for enums::AttemptStatus {
+                        charge_id: None,
+                    })
+                };
+                Ok(Self {
+                    status,
+                    response,
+                    ..item.data
+                })
+            }
+        }
+    }
+}
+
+#[derive(Debug, Serialize)]
+pub struct BraintreePSyncRequest {
+    query: String,
+}
+
+impl TryFrom<&types::PaymentsSyncRouterData> for BraintreePSyncRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsSyncRouterData) -> Result<Self, Self::Error> {
+        let transaction_id = item
+            .request
+            .connector_transaction_id
+            .get_connector_transaction_id()
+            .change_context(errors::ConnectorError::MissingConnectorTransactionID)?;
+        let query = format!("query {{ search {{ transactions(input: {{ id: {{is: \"{}\"}} }}, first: 1) {{ edges {{ node {{ id status createdAt amount {{ value currencyCode }} orderId }} }} }} }} }}", transaction_id);
+        Ok(Self { query })
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct NodeData {
+    id: String,
+    status: BraintreePaymentStatus,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct EdgeData {
+    node: NodeData,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct TransactionData {
+    edges: Vec<EdgeData>,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct SearchData {
+    transactions: TransactionData,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct PSyncResponseData {
+    search: SearchData,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum BraintreePSyncResponse {
+    PSyncResponse(Box<PSyncResponse>),
+    ErrorResponse(Box<ErrorResponse>),
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct PSyncResponse {
+    data: PSyncResponseData,
+}
+
+impl<F, T>
+    TryFrom<types::ResponseRouterData<F, BraintreePSyncResponse, T, types::PaymentsResponseData>>
+    for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<F, BraintreePSyncResponse, T, types::PaymentsResponseData>,
+    ) -> Result<Self, Self::Error> {
+        match item.response {
+            BraintreePSyncResponse::ErrorResponse(error_response) => Ok(Self {
+                response: build_error_response(&error_response.errors, item.http_code),
+                ..item.data
+            }),
+            BraintreePSyncResponse::PSyncResponse(psync_response) => {
+                let edge_data = psync_response
+                    .data
+                    .search
+                    .transactions
+                    .edges
+                    .first()
+                    .ok_or(errors::ConnectorError::MissingConnectorTransactionID)?;
+                let transaction_id = &edge_data.node.id;
```

Comment:
Do we need to consume the transaction_id in the PSync flow?

Reply:
Not needed , `&edge_data.node.id;` is returning the same transaction id we get while doing the actual payment
removed